### PR TITLE
fix build, update dependencies

### LINKED ISSUE #285: HEAD doesn't build
Verified both locally (OSX) and on Shippable and Travis (Linux). Error is in the "concat:full" task, and looks very similar to the issue that lead me to pin `grunt-browserify` (https://github.com/freedomjs/freedom/issues/278) - namely, `Error code: ENAMETOOLONG`.

HEAD used to build ~10 days ago and hasn't changed since, so the issue is some latent dependency shifting. Will research, but obviously the short term fix would be to pin more stuff.

### PR DESCRIPTION
**Pull request by soycode**
Fixes #285, #278, and more definitively resolves the issue first discussed in #276.

This takes a rather brute but hopefully robust-to-latent-dependency-changes approach to fixing our build issue. `concat` doesn't play nicely with sourcemaps-as-files that have unicode, but we don't need any unicode (our codebase and generated files are entirely ascii) so we can just remove the `charset:utf-8` clause and things work as before. Obviously ideally we'll go back and remove this replacement approach once dependencies are more cooperative - I'll open a tracking issue so we don't lose track, but think that at the moment more investigation would be more cost than benefit.

The other changes in this request are just to bump dependencies and smooth over the few spots where that means changing things (of note, junit reporter now takes an outputDir rather than file).

### CODE REVIEW

**Comment by soycode** [comment]
@ryscheng @willscott @dborkan PTAL - not urgent in and of itself, but getting this (or something similar) merged will be blocking for getting other changes out (e.g. cutting a release for Dan's changes). Thanks!

File: Gruntfile.js [hunk 1]
```diff
@@ -188,7 +185,19 @@ module.exports = function (grunt) {
     // Exorcise, Uglify, Concat are used to create a minimized freedom.js with
     // correct sourcemap. Uglify needs an explicit 'sourceMapIn' argument,
     // requiring that exorcise be used before hand. Concat is able to properly
-    // attach a banner while maintaining the correct source-map offsets.
+    // attach a banner while maintaining the correct source-map offsets.    
+    // The replace tasks removes a charset declaration that breaks concat -
+    // it should be removed once concat copes better.
+    replace: {
```
Comment (willscott) [inline Gruntfile.js]: would be good to verify that source-map offsets make sense. otherwise, looks good!

Comment (soycode) [inline Gruntfile.js]: Loading up the generated freedom.min.js(.map) in Chrome and playing with it a bit, it looks like line numbers correspond appropriately. Not sure if you were envisioning/intending some other check, let me know - thanks!

Comment (willscott) [inline Gruntfile.js]: Just making sure that break points line up and can be set in the source-map
files appropriately, or if they end up breaking a line off.  if things seem
good, then this sounds good.

Comment (soycode) [inline Gruntfile.js]: K yeah at least from a bit more ad hoc investigation I'd say things look lined up. LMK if there's a specific check you'd like.

**Pull request merged**